include/spirv-tools/optimizer.hpp

```diff
@@ -185,6 +185,15 @@ Optimizer::PassToken CreateUnifyConstantPass();
 // OpSpecConstantOp.
 Optimizer::PassToken CreateEliminateDeadConstantPass();
 
+// Creates a block merge pass.
+// This pass searches for blocks with a single Branch to a block with no
+// other predecessors. It merges the blocks into a single block.
```

Comment:
Re OpPhi and OpName, these feel almost more like bugs rather than design. Neither of these seem to have a legitimate reason to stop this optimization. I will probably just add code to remove these restrictions rather than document them.